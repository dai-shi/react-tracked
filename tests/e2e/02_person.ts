import { describe, expect, it } from 'vitest';
import puppeteer from 'puppeteer';

describe('02_typescript', () => {
  const port = process.env.PORT || '8080';

  it('should work with recorded events', async () => {
    // eslint-disable-next-line import/no-named-as-default-member
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.goto(`http://localhost:${port}/`);

    await page.waitForSelector(
      'body > #app > div:nth-child(2) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(2) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(2) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(2) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(2) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(2) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(2) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(2) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      '#app > div:nth-child(5) > div > div > button:nth-child(2)',
    );
    await page.click(
      '#app > div:nth-child(5) > div > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      '#app > div:nth-child(5) > div > div > button:nth-child(2)',
    );
    await page.click(
      '#app > div:nth-child(5) > div > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      '#app > div:nth-child(5) > div > div > button:nth-child(3)',
    );
    await page.click(
      '#app > div:nth-child(5) > div > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      '#app > div:nth-child(5) > div > div > button:nth-child(3)',
    );
    await page.click(
      '#app > div:nth-child(5) > div > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(5) > div:nth-child(2) > input',
    );
    await page.click(
      'body > #app > div:nth-child(5) > div:nth-child(2) > input',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(2) > input',
      'a',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(2) > input',
      'b',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(2) > input',
      'c',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(3) > input',
      'd',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(3) > input',
      'e',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(3) > input',
      'f',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(4) > input',
      '1',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(4) > input',
      '2',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(4) > input',
      '3',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await browser.close();
  });
});
