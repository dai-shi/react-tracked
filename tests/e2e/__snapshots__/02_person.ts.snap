// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`02_typescript > should work with recorded events 1`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 18<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 20<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>numRendered: 36<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 6<div>numRendered: 40<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 2`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 26<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 28<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>numRendered: 52<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 6<div>numRendered: 56<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 3`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 34<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 36<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>numRendered: 68<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 6<div>numRendered: 72<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 4`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 42<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 44<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>numRendered: 84<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 6<div>numRendered: 88<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 5`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 50<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 52<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>numRendered: 100<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 6<div>numRendered: 104<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 6`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 58<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 60<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>numRendered: 116<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 6<div>numRendered: 120<div><span>Count: 2</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 7`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 66<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 68<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>numRendered: 132<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 6<div>numRendered: 136<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 8`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 74<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 76<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>numRendered: 148<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 6<div>numRendered: 152<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 9`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 74<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 76<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>numRendered: 148<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 6<div>numRendered: 152<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 10`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 74<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 76<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 162<div>numRendered: 164a<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="a"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 166<div>numRendered: 168a<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="a"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 11`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 74<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 76<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 178<div>numRendered: 180ab<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="ab"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 182<div>numRendered: 184ab<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="ab"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 12`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 74<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 76<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 194<div>numRendered: 196abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 198<div>numRendered: 200abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 13`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 74<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 76<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 210<div>numRendered: 212abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="d"></div><div>Age:<input value="0"></div></div><div>numRendered: 214<div>numRendered: 216abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="d"></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 14`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 74<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 76<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 226<div>numRendered: 228abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="de"></div><div>Age:<input value="0"></div></div><div>numRendered: 230<div>numRendered: 232abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="de"></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 15`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 74<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 76<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 242<div>numRendered: 244abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="def"></div><div>Age:<input value="0"></div></div><div>numRendered: 246<div>numRendered: 248abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="def"></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 16`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 74<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 76<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 258<div>numRendered: 260abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="def"></div><div>Age:<input value="1"></div></div><div>numRendered: 262<div>numRendered: 264abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="def"></div><div>Age:<input value="1"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 17`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 74<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 76<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 274<div>numRendered: 276abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="def"></div><div>Age:<input value="12"></div></div><div>numRendered: 278<div>numRendered: 280abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="def"></div><div>Age:<input value="12"></div></div></div>
  

"
`;

exports[`02_typescript > should work with recorded events 18`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 74<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 76<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 290<div>numRendered: 292abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="def"></div><div>Age:<input value="123"></div></div><div>numRendered: 294<div>numRendered: 296abc<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>First Name:<input value="abc"></div><div>Last Name:<input value="def"></div><div>Age:<input value="123"></div></div></div>
  

"
`;
